src/docs/development/platform-integration/add-to-app/performance.md

```diff
@@ -0,0 +1,174 @@
+on the host platform is also created at this point to support the Dart Isolate,
+such as offloading GPU handling and image decoding off of the Dart Isolate.
+
+One Dart Isolate exists per `FlutterEngine` instance and multiple Dart Isolates
+can be hosted by the same Dart VM.
+
+On **Android**, this happens when you call [`DartExecutor.executeDartEntrypoint()`]({{site.api}}/javadoc/io/flutter/embedding/engine/dart/DartExecutor.html#executeDartEntrypoint-io.flutter.embedding.engine.dart.DartExecutor.DartEntrypoint-)
+on a `FlutterEngine` instance.
+
+On **iOS**, this happens when you call [`runWithEntrypoint:`]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:)
+on a `FlutterEngine`.
+
+At this point, your Dart code's selected entrypoint (the `main()` function of
+your Dart library's main.dart file by default) is executed. If you called the
+Flutter function [runApp()]({{site.api}}/flutter/widgets/runApp.html) in your
+`main()` function, your Flutter app or library's widget tree will also be
+created and built. If you need to gate certain functionalities from executing
+in your Flutter code, the `AppLifecycleState.detached` enum value will indicate
+that the `FlutterEngine` is not yet attached to any UI components such as a
+`FlutterViewController` on iOS or a `FlutterActivity` on Android.
+
+### Attaching UI to the Flutter Engine
+
+A standard full-Flutter app moves to reach this state as soon as the app is
+launched.
+
+In an add-to-app scenario, this happens when you attach a `FlutterEngine`
+to a UI component such as by calling [`startActivity()`](https://developer.android.com/reference/android/content/Context.html#startActivity(android.content.Intent))
+with an [`Intent`](https://developer.android.com/reference/android/content/Intent.html)
+built via [`FlutterActivity.withCachedEngine()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#withCachedEngine-java.lang.String-)
+on **Android**. Or by presenting a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html)
+initialized via [`initWithEngine: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithEngine:nibName:bundle:)
+on **iOS**.
+
+This would also the case if a Flutter UI component was launched without
+pre-warming a `FlutterEngine` such as via [`FlutterActivity.createDefaultIntent()`]({{site.api}}/javadoc/io/flutter/embedding/android/FlutterActivity.html#createDefaultIntent-android.content.Context-)
+on **Android** or via [`FlutterViewController initWithProject: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithProject:nibName:bundle:)
+on **iOS**. An implicit `FlutterEngine` is created in these cases. ;
+
+Behind the scene, both platform's UI components serves to provide the
```

Comment:
"both platform's UI components" ?? What two components are you talking about?

Also, if it really is two platform components, it would be "serve", not "serves".

Reply:
I meant the Android and iOS platforms' UI components